### PR TITLE
Pxweb2 179 variablebox update table

### PR DESCRIPTION
**Pull request by michaelpande**

### CODE REVIEW

File: apps/pxweb2/src/app/context/TableDataProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }, [errorMsg]);
+
+  const fetchTableData = async (tableId: string, i18n: i18n) => {
+    const valueCodes: Record<string, Array<string>> = {};
+
+    const ids = variables.getUniqueIds();
+    ids.forEach((id) => {
+      valueCodes[id] = variables.getSelectedValuesById(id);
+    });
+    console.log(valueCodes);
```
Comment (SjurSutterudSagen) [inline apps/pxweb2/src/app/context/TableDataProvider.tsx:40]: Is this left on purpose?

Comment (michaelpande) [reply apps/pxweb2/src/app/context/TableDataProvider.tsx:40]: No, I will remove it 😄

File: apps/pxweb2/src/app/context/VariablesProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    return false;
+  };
+
+  const syncVariables = (variables: SelectedVBValues[]) => {
+    if (!hasChanges(variables)) {
+      return;
+    }
+    setVariables(new Map());
+    variables.forEach((id) => {
+      addSelectedVariables(id.id, id.values);
```
[the remark this answers is not in the document]
Comment (michaelpande) [reply apps/pxweb2/src/app/context/VariablesProvider.tsx:146]: Good catch, totally agree. It should be named variable.

File: apps/pxweb2/src/app/context/VariablesProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+}) => {
+  const [variables, setVariables] = useState<
+    Map<string, { id: string; value: string }>
+  >(new Map());
+
+  const addSelectedVariable = (id: string, value: string) => {
+    setVariables((prev) => new Map(prev).set(id + '-' + value, { id, value }));
+  };
+
+  const addSelectedVariables = (id: string, values: string[]) => {
```
[the remark this answers is not in the document]
Comment (michaelpande) [reply apps/pxweb2/src/app/context/VariablesProvider.tsx:52]: Hmm... I agree, will see what I can do to improve this :)

**Review by SjurSutterudSagen — APPROVED** [review]
Looks good, good job on this one :)

**Pull request merged**